epaint/src/shape.rs

```diff
@@ -86,6 +86,36 @@ impl Shape {
         }
     }
 
+    /// Turn a line into equally spaced dots.
+    pub fn dotted_line(
+        points: &[Pos2],
+        color: impl Into<Color32>,
+        spacing: f32,
+        radius: f32,
+    ) -> Vec<Self> {
+        let mut shapes = Vec::new();
+        points_from_line(points, spacing, radius, color.into(), &mut shapes);
+        shapes
+    }
+
+    /// Turn a line into dashes.
+    pub fn dashed_line(
+        points: &[Pos2],
+        stroke: impl Into<Stroke>,
+        dash_length: f32,
+        dash_gap_ratio: f32,
```

Comment:
`dash_gap_ratio` deserves an explanation. Length of dash divided by length of the gap between them? it is a bit of a weird API imho. `dash_length` and `gap_length` seems simpler.

Reply:
Applied in 5418cd7.